### DISCUSSION TITLE
Zotero with Hyprland

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [hyprwm/Hyprland#1269] [Q&A] [answered]
I have a window rule that floats a certain window B.

On screen I already have window A maximized, when I open window B.

The problem is that when window B is opened, it is first tiled, and so window A is resized.

Then window B is floated, and window A gets maximized again.

But window A is LibreOffice Writer, and the tiling -> maximization process mess up the zoom and the position of the page.

Is there a way to float window B immediately, without first tiling it?

**Comment by raffaelemancuso** [comment]
Modified to clarify my question

**Comment by vaxerski** [comment]
it should not be like that. Are you sure you're applying the rule properly?

does e.g.
```
hyprctl dispatch exec "[float] kitty"
````
do the same?

**Reply by raffaelemancuso** [reply]
Window B is the Zotero window to insert citation.

I think the real process is this: i opens another application before opening the "real" window (and close the previous application).

How can I monitor clients?

That is, how can I monitor changes of `hyprctl clients` over time?

**Comment by raffaelemancuso** [comment, marked as answer]
When you add a citation in LibreOffice, in reality Zotero opens two windows.

The first window is a sort of "Loading..." window, titled "Progress", while the second window is the real input that asks for user input, and is titled "Quick Format Citation".

So to have a smooth experience you need to apply the rules to both of them.

This is my ruleset to use Zotero with Hyprland, in case somebody else needs it:
```
windowrulev2 = float,class:^(Zotero)$,title:^(Quick Format Citation|Progress)$
windowrulev2 = center,class:^(Zotero)$,title:^(Quick Format Citation|Progress)$
windowrulev2 = noanim,class:^(Zotero)$,title:^(Quick Format Citation|Progress)$
windowrulev2 = maxsize 400 100,class:^(Zotero)$,title:^(Quick Format Citation|Progress)$
```

**Reply by kescobo** [reply]
This doesn't seem to work (anymore?) - at least, the citation window is centered as expected, but the "dropdown" that shows matching documents remains invisible. 

<img width="930" height="246" alt="image" src="https://github.com/user-attachments/assets/dd58cdbe-ecac-492d-be95-95db95c542db" />

Whereas if I maximize the window:

<img width="828" height="413" alt="image" src="https://github.com/user-attachments/assets/e58fe7c3-adc3-46c3-8bfa-0a007976fba4" />

Seems like I must be missing something...

**Reply by kescobo** [reply]
Updating to 7.1-beta and using the rules [suggested here](https://github.com/hyprwm/Hyprland/discussions/10287#discussioncomment-13076678) worked for me

```
# Allow zotero cite windows to float
windowrulev2 = float,class:^(Zotero)$,title:^($|Citation Dialog$)
windowrulev2 = center,class:^(Zotero)$,title:^($|Citation Dialog$)
```
